patterns.csv

```diff
@@ -1,8 +1,8 @@
 "Pattern"	"Label"	"Description"
-"[A-Z0-9]{2}-[A-Z0-9]{2}-[A-Z0-9]{2}"	"MAT"	"Matriculas portuguesas 0A-0A-0A"
+"(?<=[^\w-])(?:[A-Z]{2}-\d{2}-\d{2}|[A-Z]{2}-\d{2}-[A-Z]{2}|\d{2}-[A-Z]{2}-\d{2}|\d{2}-\d{2}-[A-Z]{2}|[A-Z]{2}\s?\d{2}\s?[A-Z]{2})(?![A-Z\d])(?=[^\w-])"	"MAT"	"Matriculas portuguesas 0A-0A-0A"
 "\d+(\-\w+)?\/(\w+\.)(\w+\.?)+"	"PRO"	"Processos com / (NN/...) opcionalmente apanha (NN-W/...) até ao próximo espaço"
-"\b\d{1,2}(-|\.|/)\d{1,2}(-|\.|/)\d{2,4}\b"	"DAT"	"Datas NN.NN.NNNN ou NN-NN-NNNN ou NN/NN/NNNN"
+"0?[1-31]{1,2}[-\/][1-12]{1,2}[-\/][1-2023]{2,4}"	"DAT"	"Datas NN.NN.NNNN ou NN-NN-NNNN ou NN/NN/NNNN"
```

Comment:
I get what you are trying to do here, but this doesn't work as you think it would.
[X-Y] Means all characters between X and Y. [NX-MY] Means character N, all characters between X and M and character Y. Which means that any of the following dates (with character 4 or higher) are failing:
 - 06/06/1997
 - 05/09/2023
 - 05/09/2024

Test cases: https://regex101.com/r/n8ZE7a/1

Reply:
Thanks for the review! Here is the updated regex rule "(?<=[^\w-])(?:[A-Z]{2}-\d{2}-\d{2}|[A-Z]{2}-\d{2}-[A-Z]{2}|\d{2}-[A-Z]{2}-\d{2}|\d{2}-\d{2}-[A-Z]{2}|[A-Z]{2}\s?\d{2}\s?[A-Z]{2}|\d{1,2}/\d{1,2}/\d{2,4}|\d{1,2}-\d{1,2}-\d{2,4})(?![A-Z\d])(?=[^\w-])"